src/tests.rs

```diff
@@ -94,3 +96,181 @@ async fn test_disputes_offchain_disabling() -> Result<(), Error> {
+
+    // Netowrk is still healthy
+    assert_blocks_are_being_finalized(&client).await?;
+
+    // get runtime version before the upgrade
+    let version_before = get_runtime_version(&client).await?;
+
+    // runtime upgrade
+    let code = std::fs::read(
+        Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("artifacts/westend_runtime.compact.compressed.wasm"),
+    )?;
+
+    println!("Starting runtime upgrade");
+    perform_runtime_upgrade(&client, code).await?;
+    println!("🚀🚀🚀 runtime upgrade complete");
+
+    assert_blocks_are_being_finalized(&client).await?;
+
+    // get runtime version after the upgrade - is it newer?
+    let version_after = get_runtime_version(&client).await?;
+    assert!(version_after.spec_version > version_before.spec_version);
+
+    // still no disputes
+    wait_for_metric(honest, DISPUTES_CONCLUDED_VALID, 0).await?;
+
+    Ok(())
+}
+
+async fn assert_blocks_are_being_finalized(
+    client: &OnlineClient<PolkadotConfig>,
+) -> Result<(), Error> {
+    let mut finalized_blocks = client.blocks().subscribe_finalized().await?;
+    let first_measurement = finalized_blocks
+        .next()
+        .await
+        .ok_or(Error::from("Can't get finalized block from stream"))??
+        .number();
+    sleep(Duration::from_secs(30)).await;
+    let second_measurement = finalized_blocks
+        .next()
+        .await
+        .ok_or(Error::from("Can't get finalized block from stream"))??
+        .number();
+
+    assert!(second_measurement > first_measurement);
+
+    Ok(())
+}
+
+// This function contains hacks.
+// At the moment zombienet doesn't support restarting a process with a new binary. As a work around
+// system PATH is modified (by prepending a new location) before restarting the processes. This way
+// zombienet will start the new binary and perform a 'client update'.
+// The function also expects 'the new binaries' to be located in `artifacts/polkadot-disabling`.
+// Building a set of binaries can be done with:
+// `cargo build --profile testnet --features pyroscope,fast-runtime \
+// -p test-parachain-adder-collator -p test-parachain-undying-collator -p polkadot-test-malus -p polkadot -p polkadot-parachain-bin`.
+// List of the required binaries:
+//  * target/testnet/adder-collator
+//  * target/testnet/malus
+//  * target/testnet/polkadot
+//  * target/testnet/polkadot-execute-worker
+//  * target/testnet/polkadot-prepare-worker
+//  * target/testnet/undying-collator
+//  * target/testnet/polkadot-parachain
+pub async fn perform_nodes_upgrade(network: &Network<LocalFileSystem>) -> Result<(), Error> {
+    prepend_to_system_path(
+        Path::new(env!("CARGO_MANIFEST_DIR"))
+            .join("artifacts/polkadot-disabling")
+            .as_path(),
+    )
+    .await?;
+
+    println!("Starting nodes update");
+
+    for node in network.nodes() {
+        node.restart(Some(Duration::from_secs(5))).await?;
+    }
+
+    println!("🚀🚀🚀 nodes update complete");
+
```

Comment:
this will probably go away once we migrate these tests to CI, but wouldn't starting the binaries with absolute path like 
```rust
        Path::new(env!("CARGO_MANIFEST_DIR"))
            .join("artifacts/polkadot-disabling")
            .as_path()
```
avoid the need of this hack?

Reply:
If I understand correctly you are suggesting to use the full path when starting the network ([here](https://github.com/paritytech/disabling-e2e-tests/pull/6/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R134))?

If we use the full path, the hack with the path won't work and we'll have to switch binaries. Not that it is hard to do it but it's again a hack.

Once `restart` starts supporting `binary` parameter we can use full paths on both places and get rid of this hack.

Reply:
yeah, I was thinking that swapping local binaries (or changing a symlink) is a smaller hack than modifying the global `$PATH` variable, but we don't need to worry about it for now